tests/test_dmxrange_add_sub.py

```diff
@@ -0,0 +1,140 @@
+"""Test functions for adding and removing DMX ranges."""
+
+import pytest
+import io
+import numpy as np
+from pint.models import get_model
+from pint.models.dispersion_model import DispersionDM, DispersionDMX
+
+
+def test_startMJD_greaterThan_endMJD():
+    """ Check for error when start MJD is larger than end MJD. """
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58100, 58000, 3, 1, frozen=False)
+
+
+def test_only_one_MJD():
+    """ Check for error when one of the MJDs is None and the other isn't. """
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(None, 58000, 3, 1, frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58000, None, 3, 1, frozen=False)
+
+
+def test_duplicate_index():
+    """ Check for error when a duplicate DMX index is used. """
+    dm_mod = DispersionDMX()
+    dm_mod.add_DMX_range(58000, 58100, 3, 1, frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58200, 58300, 3, 1, frozen=False)
+
+
+def test_remove_nonexistent_index():
+    """ Check for error when a unused DMX index is removed. """
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.remove_DMX_range(3)
+
+
+def test_nonetype_MJD():
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(None, 58000, 3, 1, frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58000, None, 3, 1, frozen=False)
+
+
+def test_dynamic_index():
+    dm_mod = DispersionDMX()
+    index1 = 3
+    dmx1 = 1.0
+    mjd_start1 = 58000.0
+    mjd_end1 = 58100.0
+    index2 = None
+    dmx2 = 1.0
+    mjd_start2 = 58200.0
+    mjd_end2 = 58300.0
+    dm_mod.add_DMX_range(mjd_start1, mjd_end1, index1, dmx1, frozen=False)
+    dct = dm_mod.get_prefix_mapping_component("DMX_")
+    ind = np.max(list(dct.keys())) + 1
+    dm_mod.add_DMX_range(mjd_start2, mjd_end2, index2, dmx2, frozen=False)
+    dct = dm_mod.get_prefix_mapping_component("DMX_")
+    assert np.max(list(dct.keys())) == ind
```

Comment:
This test requires a specific algorithm for choosing the dynamic indices. Better to check that the returned index corresponds to a real parameter and was not previously in the set of indices.

Reply:
okay i changed this a bit (i was forgetting that i had set it up to return the index of the new DMX range when one is added). let me know if it seems good enough now.